src/aosm/azext_aosm/deploy/deploy_with_arm.py

```diff
@@ -208,6 +246,134 @@ def construct_manifest_parameters(self) -> Dict[str, Any]:
+            )
+
+        if parameters_json_file:
+            message = f"Use parameters from file {parameters_json_file}"
+            logger.info(message)
+            print(message)
+            with open(parameters_json_file, "r", encoding="utf-8") as f:
+                parameters = json.loads(f.read())
+
+        else:
+            # User has not passed in parameters file, so we use the parameters required
+            # from config for the default bicep template produced from building the
+            # NFDV using this CLI
+            logger.debug("Create parameters for default NFDV template.")
+            parameters = self.construct_cnfd_parameters()
+
+        logger.debug(
+            f"Parameters used for CNF definition bicep deployment: {parameters}"
+        )
+
+        # Create or check required resources
+        deploy_manifest_template = not self.nfd_predeploy(definition_type=CNF)
+        if deploy_manifest_template:
+            self.deploy_manifest_template(
+                manifest_parameters_json_file, manifest_bicep_path, CNF
+            )
+        else:
+            print(
+                f"Artifact manifests exist for NFD {self.config.nf_name} "
+                f"version {self.config.version}"
+            )
+        message = (
+            f"Deploy bicep template for NFD {self.config.nf_name} version {self.config.version} "
+            f"into {self.config.publisher_resource_group_name} under publisher "
+            f"{self.config.publisher_name}"
+        )
+        print(message)
+        logger.info(message)
+        self.deploy_bicep_template(bicep_path, parameters)
+        print(f"Deployed NFD {self.config.nf_name} version {self.config.version}.")
+
+        acr_properties = self.api_clients.aosm_client.artifact_stores.get(
+            resource_group_name=self.config.publisher_resource_group_name,
+            publisher_name=self.config.publisher_name,
+            artifact_store_name=self.config.acr_artifact_store_name,
+        )
+        target_registry_name = acr_properties.storage_resource_id.split("/")[-1]
+        target_registry_resource_group_name = acr_properties.storage_resource_id.split(
+            "/"
+        )[-5]
+
+        acr_manifest = ArtifactManifestOperator(
+            self.config,
+            self.api_clients,
+            self.config.acr_artifact_store_name,
+            self.config.acr_manifest_name,
+        )
+
+        artifact_dictionary = {}
+
+        for artifact in acr_manifest.artifacts:
+            artifact_dictionary[artifact.artifact_name] = artifact
+
+        for package_number in range(len(self.config.helm_packages)):
+            helm_package_name = self.config.helm_packages[package_number].name
+
+            if helm_package_name not in artifact_dictionary.keys():
+                print(
+                    f"Artifact {helm_package_name} not found in the artifact manifest"
+                )
+                continue
```

Comment:
Do we ever expect to hit this?  Would it be an error?

Reply:
If the autogenerated bicep templates have not been modified by the user then we do not expect to hit this. If we do hit it it means that we have some discrepancy between the bicep templates and the input.json file. I decided not to raise an error and just skip the helm package that is not in the artifact manifest, but I can raise an error if you want.

Reply:
I think it is better to raise an error and say what the discrepancy is.